plugin/cog/cog-rangereader-azure/src/main/java/it/geosolutions/imageioimpl/plugins/cog/AzureConfigurationProperties.java

```diff
@@ -52,47 +51,46 @@ public class AzureConfigurationProperties {
 
         if (path.startsWith("https")) {
             // replace for quicker finding
-            path = path.replace("https", "http");
+            path = path.replaceFirst("https", "http");
             useHTTPS = true;
         }
 
-        BlobURLParts parts = null;
         try {
-            parts = URLParser.parse(uri.toURL());
-            container = parts.containerName();
-            String blobName = parts.blobName();
+            BlobUrlParts parts = BlobUrlParts.parse(uri.toURL());
+            container = parts.getBlobContainerName();
+            String blobName = parts.getBlobName();
             int lastIndex = blobName.lastIndexOf("/");
             if (lastIndex > 0) {
                 prefix = blobName.substring(0, lastIndex);
             }
-            String host = parts.host();
+            String host = parts.getHost();
             int blobcoreIdx = host.indexOf("." + AzureClient.AZURE_URL_BASE);
             if (blobcoreIdx > 0) {
                 accountName = host.substring(0, blobcoreIdx);
             }
 
-        } catch (UnknownHostException| MalformedURLException e) {
+        } catch (IllegalStateException| MalformedURLException e) {
             throw new RuntimeException("Unable to parse the provided uri " + path + "due to " + e.getLocalizedMessage());
         }
 
-        if (container == null) {
+        if (container == null) {//REVISIT: dead code
             container = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_CONTAINER, null);
         }
-        if (prefix == null) {
+        if (prefix == null) {//REVISIT: dead code
             prefix = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_PREFIX, null);
         }
         if (cogUri.getUser() != null && cogUri.getPassword()!= null) {
             accountName = cogUri.getUser();
             accountKey = cogUri.getPassword();
         }
 
-        if (accountName == null) {
+        if (accountName == null) {//REVISIT: dead code
             accountName = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_NAME, null);
         }
         if (accountKey == null) {
             accountKey = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_KEY, null);
         }
-        if (maxConnections == null) {
+        if (maxConnections == null) {//REVISIT: dead code
```

Comment:
However for example this one... why is it dead code?

Reply:
Cause it's in the constructor and the variable is initialized
```
private Integer maxConnections = 64;
```

Reply:
If you can prove some code is dead, just remove it then.

Reply:
Still the intention is you could configure it with the env variables/system props, hence the revisit note. I'd rather keep the pr focused